kart/geometry.py

```diff
@@ -76,14 +113,32 @@ def crs_id(self):
 
     @property
     def geometry_type(self):
-        flags = _validate_gpkg_geom(self)
-        envelope_size = gpkg_envelope_size(flags)
+        envelope_size = gpkg_envelope_size(self.flags)
         wkb_offset = 8 + envelope_size
         wkb_is_le, geom_type = _wkb_endianness_and_geometry_type(
             self, wkb_offset=wkb_offset
         )
         return geom_type
 
+    @property
+    def envelope(self):
```

Comment:
Yup done. Rob's one is a lot more cautious than mine - mine assumes that Kart geometries are V1, are not extended, have been normalised (which is more likely to be true now that we make an effort to check for schema violations and normalise geometries during commits). But, probably still better to use the cautious code. Extended it slightly so that it can return 2D envelopes or whatever it finds, and, so that using OGR to generate when missing can be turned off.